### PR TITLE
fix: svg morphing

### PR DESCRIPTION
**Pull request by amymc**
https://github.com/viewstools/morph/issues/52

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -115,15 +120,30 @@ if (shouldWatch) {
...
+
+        resolve(
+          onMorphWriteSvg({
+            as,
+            code: inlined,
+            isInlineSvg: true,
+            file: path.resolve(input, '..', `${pathToName(input)}.view`),
+          })
+        )
+      })
```
Comment (dariocravero) [inline cli.js]: Is there any reason why you wrapped the whole thing in a promise? it seems to me that you can get away with doing something like:
```
morphInlineSvg(input).then(code => writeFile(path.resolve(input, '..', `${pathToName(input)}.view`), code)
```

Also, when we morph a view in the next bit of logic below, we're outputting the code to the console, so you can do like in unix tools and pipe it to something else like a file or copy it from the terminal to something else. Ideally we'd keep the two of them consistent. What're your thoughts on this?

Thanks for bringing this back 😄

**Comment by dariocravero** [comment]
Oh, one thing I just remembered to double check, is the morpher taking into account the new proximity nesting syntax changes you made a few days ago?

**Comment by amymc** [comment]
Hey, re the proximity nesting, I know we talked about it earlier but now that I think about it, is it relevant here? If we're morphing an svg file, it's just going to be the svg, there will never be anything after it, so we won't need the blank line after it. What do you think?

**Comment by dariocravero** [comment]
You're right in that you only morph the SVG on its own but the thing is that before the even proximity nesting, you had this:
```
Svg
P1 SvgPath
P2 SvgPath
G1 SvgGroup
G1P1 SvgPath
G1P2 SvgPath

P3 SvgPath
```

and after it would be:
```
Svg
P1 SvgPath

P2 SvgPath

G1 SvgGroup
G1P1 SvgPath

G1P2 SvgPath


P3 SvgPath
```

Because only `Svg`, `SvgGroup` and a few others [are groups](https://github.com/viewstools/morph/blob/master/parse/helpers.js#L62), those only needed a new line on the previous morpher but now all of them do.

Does it make sense?

**Comment by amymc** [comment]
Yeah, got it. Thanks!

**Comment by amymc** [comment]
I think I've covered all the bits we talked about on Friday.

- Props have been replaced with slots
- The `||` on width and height are gone
- If width and height are not declared, they are inferred from the viewbox
- Fill and stroke are exposed as slots
- Multiples of the same attribute are exposed as `fill2`, `fill3` etc.

**Comment by dariocravero** [comment]
Amazing Amy!! Thanks so much! Will have a proper look at it today :)

**Comment by amymc** [comment]
Sorry it took me a while to get back to this. I got a bit sidetracked by the chrome extension :grimacing:
It's a while since we talked about this so I'm not sure if I've implemented this the way we discussed :see_no_evil:

**Comment by dariocravero** [comment]
Amazing @amymc! Thanks a mill for doing this! :) It's released on v12.1.0!

**Pull request merged**